lib/msal-native-auth/src/controller/NativeAuthStandardController.ts

```diff
@@ -0,0 +1,229 @@
+
+            // start the signin flow
+            const signInStartParams: SignInStartParams = new SignInStartParams(
+                authorityUrl,
+                this.config.auth.clientId,
+                correlationId,
+                this.nativeAuthConfig.nativeAuth.challengeTypes ?? [],
+                signInOptions.scopes ?? [],
+                signInOptions.username,
+                signInOptions.password
+            );
+
+            const startResult = await this.signInClient.start(
+                signInStartParams
+            );
+
+            if (startResult instanceof SignInWithContinuationTokenResult) {
+                // require password
+                if (!signInOptions.password) {
+                    return new SignInResult(
+                        undefined,
+                        new SignInPasswordRequiredStateHandler(
+                            this.signInClient,
+                            correlationId,
+                            startResult.continuationToken,
+                            this.nativeAuthConfig,
+                            signInOptions.scopes
+                        )
+                    );
+                }
+
+                // if the password is provided, then try to get token silently.
+                const signInSubmitPasswordParams =
+                    new SignInSubmitPasswordParams(
+                        authorityUrl,
+                        this.config.auth.clientId,
+                        correlationId,
+                        this.nativeAuthConfig.nativeAuth.challengeTypes ?? [],
+                        signInOptions.scopes ?? [],
+                        startResult.continuationToken,
+                        signInOptions.password
+                    );
+
+                const completedResult = await this.signInClient.submitPassword(
+                    signInSubmitPasswordParams
+                );
+
+                const accountManager = new AccountInfo(
+                    completedResult.authenticationResult.account,
+                    correlationId,
+                    this.nativeAuthConfig
+                );
+
+                return new SignInResult(accountManager);
+            } else if (startResult instanceof SignInCodeSendResponse) {
+                // require code
+                return new SignInResult(
+                    undefined,
+                    new SignInCodeRequiredStateHandler(
+                        this.signInClient,
+                        correlationId,
+                        startResult.continuationToken,
+                        this.nativeAuthConfig,
+                        signInOptions.scopes
+                    )
+                );
+            } else {
+                throw new UnexpectedError("Unknow SignInStartResult type");
+            }
+        } catch (error) {
+            return SignInResult.createWithError(error);
+        }
+    }
```

Comment:
My suggestion here is to refactor this method because there are early returns disconected from each other, making the code hard to read and test.